eloquentjs/2_2_fizzbuzz.js

```diff
@@ -0,0 +1,26 @@
+/** Performs the FizzBuzz operation from 1 to iter(inclusive)*/
+function fizzBuzz(iter) {
+    if (isNaN(iter)) {
+        throw "FizzBuzz input is NaN";
+        return;
+    }
+    if (iter <= 0) {
+        throw "FizzBuzz input is less than zero";
+        return;
+    }
+    for (let i = 1; i <= iter; i++) {
+        let output = '';
```

Comment:
The pattern of declaring the a variable and then building it up before you do the thing with it is powerful, but I think you overuse it a bit. It has the downside that now you have a mutable value you have to chase around, which makes editing the code in the future more prone to issues. I'd recommend solving this with the more immutable approach:

```js
const divisBy3 = i % 3 === 0;
const divisBy5 = i % 3 === 0;

if (divisBy3 && divisBy5) {
  console.log("FizzBuzz");
} else if (divisBy3) {
  console.log("Fizz");
} else if (divisBy5) {
  console.log("Buzz");
} else {
  console.log(i);
}
  ```